benchmarkapps/PlatformBenchmarks/BenchmarkApplication.HttpConnection.cs

```diff
@@ -81,33 +94,31 @@ private bool ParseHttpRequest(ref ReadResult result)
                     {
                         state = State.Headers;
                     }
+
+                    buffer = buffer.Slice(consumed);
                 }
 
                 if (state == State.Headers)
                 {
-                    if (parsingStartLine)
```

Comment:
Remove the var

Reply:
And use switch?

Reply:
Removed

Falls through; can't do that with switch (in C#)

Is better in follow up PR https://github.com/aspnet/KestrelHttpServer/pull/2576

```csharp
private State ParseHttpRequest(State state, ref ReadOnlySequence<byte> buffer, ref SequencePosition examined)
{
    SequencePosition consumed = default;

    if (state == State.StartLine)
    {
        if (Parser.ParseRequestLine(new ParsingAdapter(this), buffer, out consumed, out examined))
        {
            state = State.Headers;
        }

        buffer = buffer.Slice(consumed);
    }

    if (state == State.Headers)
    {
        if (Parser.ParseHeaders(new ParsingAdapter(this), buffer, out consumed, out examined, out int consumedBytes))
        {
            state = State.Body;
        }

        buffer = buffer.Slice(consumed);
    }

    return state;
}
```